applications/FreeSurfaceApplication/python_scripts/edgebased_levelset_solver.py

```diff
@@ -0,0 +1,350 @@
+from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+# importing the Kratos Library
+from KratosMultiphysics import *
+from KratosMultiphysics.FreeSurfaceApplication import *
+# from KratosMultiphysics.MeshingApplication import *
+CheckForPreviousImport()
+
+
+def AddVariables(model_part):
+    model_part.AddNodalSolutionStepVariable(VELOCITY)
+    model_part.AddNodalSolutionStepVariable(PRESSURE)
+    model_part.AddNodalSolutionStepVariable(NORMAL)
+    model_part.AddNodalSolutionStepVariable(AUX_INDEX)
+    model_part.AddNodalSolutionStepVariable(DISTANCE)
+    model_part.AddNodalSolutionStepVariable(PRESS_PROJ)
+    model_part.AddNodalSolutionStepVariable(POROSITY)
+    model_part.AddNodalSolutionStepVariable(VISCOSITY)
+    model_part.AddNodalSolutionStepVariable(DIAMETER)
+    model_part.AddNodalSolutionStepVariable(LIN_DARCY_COEF)
+    model_part.AddNodalSolutionStepVariable(NONLIN_DARCY_COEF)
+    model_part.AddNodalSolutionStepVariable(NODAL_AREA)
+    model_part.AddNodalSolutionStepVariable(STRUCTURE_VELOCITY)
+
+    print("variables for the edgebased incompressible fluid solver added correctly")
+
+
+def AddDofs(model_part):
+    for node in model_part.Nodes:
+        node.AddDof(PRESSURE)
```

Comment:
Be careful, if the formulation is implicit, with the recent changes it will give an error, because the B&S compyes always the reaction

Reply:
*Computes